src/Text/index.js

```diff
@@ -21,17 +21,26 @@ const Text = styled.span`
 `;
 
 Text.propTypes = {
-	bold: PropTypes.bool,
+	bold: PropTypes.oneOfType([PropTypes.bool, PropTypes.array]),
 	caps: PropTypes.oneOfType([
-		PropTypes.oneOf(['all', 'first']),
-		PropTypes.bool,
+		PropTypes.oneOf(['all', 'first', 'none']),
+		PropTypes.array,
+	]),
+	dimmed: PropTypes.oneOfType([PropTypes.bool, PropTypes.array]),
+	display: PropTypes.oneOfType([
+		PropTypes.oneOf(['block', 'inline', 'inline-block']),
+		PropTypes.array,
+	]),
+	fontSize: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
+	lineHeight: PropTypes.oneOfType([PropTypes.number, PropTypes.array]),
+	overflow: PropTypes.oneOfType([
+		PropTypes.oneOf(['ellipsis', 'wrap']),
+		PropTypes.array,
+	]),
+	textAlign: PropTypes.oneOfType([
```

Comment:
All of these prop definitions are shared with the above Box component. It might be nice to think about having a dictionary of prop definitions to standardize some of the common ones which many of the components like Text and Box share.

Reply:
Indeed. They would also be more accurately defined as
```js
const caseProps = PropTypes.oneOf(['all', 'first', 'none']);
...
caps: PropTypes.oneOfType([
	caseProps,
	PropTypes.arrayOf(caseProps),
])
```

And I actually tried that (and getting the caseProps from the styleProps file), but for some reason referencing an external PropTypes object breaks the propType detection, and also prevented docz (or more accurately, docGenParser), from picking up those props. I didn't want to sacrifice the props table in the docs again, so I went with the copy/paste solution. But it's not really satisfactory and I have to look into it a bit closer at some point.